### PR TITLE
Py3k

### PR DESCRIPTION
**Pull request by ocefpaf**
@ChrisBarker-NOAA we need this PR to build a Python3 version of this package.

If you accept this PR can you issue a new release and ping me?

Thanks!

### CODE REVIEW

File: unit_conversion/unit_conversion.py [hunk 1]
```diff
@@ -348,7 +352,8 @@ class InvalidUnitError(UnitConversionError):
     Exception raised when a unit is not in the Unit conversion database
 
     """
-    def __init__(self, (unit, unit_type) ):
+    def __init__(self, unit_unit_type):
+        (unit, unit_type) = unit_unit_type
```
Comment (ChrisBarker-NOAA) [inline unit_conversion/unit_conversion.py]: why is this needed? does py3 not unpack tuples in funcition signatures in the same way?

though it doesn't change the API, so no big deal either way.

File: unit_conversion/unit_conversion.py [hunk 2]
```diff
@@ -365,18 +370,20 @@ def __init__(self, unitType):
...
+
 class MismatchedUnitError(UnitConversionError):
     """
     Exception raised when a unit is not in the Unitconversion database
 
     """
     ##fixme -- this appear to be used anywhere...
-    def __init__(self, (FromUnit, FromUnitType, ToUnit, ToUnitType) ):
-        self.FromUnit     =  FromUnit    
+    def __init__(self, FromUnit_FromUnitType_ToUnit_ToUnitType):
```
Comment (ChrisBarker-NOAA) [inline unit_conversion/unit_conversion.py]: same as previous comment -- I like the old way, 'cause it makes the API clearer...

File: unit_conversion/lat_long.py [hunk 3]
```diff
@@ -267,14 +267,14 @@ def format(self, style):
     def format_html(self, style):
         """
         format_html(style)
-        
+
         Backward compatibility for Quixote rlink and Pylons inews.
         """
-        return self.format(style).replace(u"\xb0", u"&deg;").encode("ascii")
+        return self.format(style).replace(u"\xb0", u"&deg;")
```
Comment (ChrisBarker-NOAA) [inline unit_conversion/lat_long.py]: it may be important that this be ascii. though maybe ascii-compatible unicode string is OK.

I'm checking with @mikeorr -- he's the one that's used this (and wrote it)

**Comment by ChrisBarker-NOAA** [comment]
A couple tiny issues -- but I"ll probably be able to merge soon.

**Comment by ChrisBarker-NOAA** [comment]
OK -- we're not using that ascii code anyway anymore -- hopefully no on else is, we'll probably delete that code anyway. But that's for another issue....

-CHB

**Comment by ChrisBarker-NOAA** [comment]
I've bumped the version number and added a tag.

Should be good to go for py3

File: unit_conversion/unit_conversion.py (hunk 1 above)
Comment (ocefpaf) [inline unit_conversion/unit_conversion.py]: > In unit_conversion/unit_conversion.py:
>
> > @@ -348,7 +352,8 @@ class InvalidUnitError(UnitConversionError):
> >      Exception raised when a unit is not in the Unit conversion database
> >
> >      """
> > -    def __init__(self, (unit, unit_type) ):
> > +    def __init__(self, unit_unit_type):
> > +        (unit, unit_type) = unit_unit_type
>
> why is this needed? does py3 not unpack tuples in funcition signatures in
the same way?

That is no longer allowed:

http://legacy.python.org/dev/peps/pep-3113/

> though it doesn't change the API, so no big deal either way.

It does look a little bit ugly though.

**Comment by ocefpaf** [comment]
> Should be good to go for py3

Thanks! I will update the conda package tomorrow.

File: unit_conversion/unit_conversion.py (hunk 1 above)
Comment (ChrisBarker-NOAA) [inline unit_conversion/unit_conversion.py]: oh well -- I"ve got t keep this in mind with py2-py3 differences.

Thanks
